extension/tests/integration_tests.rs

```diff
@@ -860,3 +860,55 @@ async fn test_cohere() {
             .unwrap();
     assert_eq!(search_results.len(), 3);
 }
+
+#[ignore]
+#[tokio::test]
+async fn test_drop_table_triggers_job_deletion() {
+    let conn = common::init_database().await;
+    let mut rng = rand::thread_rng();
+    let test_num = rng.gen_range(1..100000);
+    let test_table_name = format!("drop_test_table_{}", test_num);
+
+    common::init_test_table(&test_table_name, &conn).await;
+
+    let insert_job_query = format!(
+        "INSERT INTO vectorize.job (name, index_dist_type, transformer, search_alg, params, last_completion)
+        VALUES ('{test_table_name}', 'pgv_hsnw_cosine', 'sentence-transformers/all-MiniLM-L6-v2', 'search_algorithm', '{{}}', NOW());"
+    );
```

Comment:
The job name does not have to be the same as the table name. For example, you can create a job named `product_search_hf` on the table named `products`. The value for the `name` column in `vectorize.table` will be "product_search_hf". The schema and table name will be stored in the params object

```sql
select name, params ->> 'table' as table, params ->> 'schema' as schema 
from vectorize.job;
       name        |  table   | schema 
-------------------+----------+--------
 product_search_hf | products | public
(1 row)
```